modal_server_vllm.py

```diff
@@ -0,0 +1,424 @@
+        else:
+            tools = None
+            tools_or_functions = []
+        # Remove any code_interpreter tools remaining
+        if tools:
+            tools = [tool for tool in tools if tool.type != "code_interpreter"]
+            tools_or_functions = [
+                tool for tool in tools_or_functions if tool["type"] != "code_interpreter"
+            ]
+        prompt_token_ids = prepare_messages_for_inference(
+                        tokenizer=self.tokenizer,
+                        messages=request.messages,
+                        functions=request.functions,
+                        tools=tools,
+                        tool_choice=request.tool_choice,
+                    ).tolist()[0]
+        error_check_ret = await check_length(request, prompt_token_ids, self.engine_model_config)
+        if error_check_ret is not None:
+            return error_check_ret
+
+        model_name = request.model
+        request_id = f"cmpl-{random_uuid()}"
+        created_time = int(time.time())
+
+        # compute stop_token_ids
+        stop_token_ids = []
+        prompt_template = get_prompt_template_from_tokenizer(self.tokenizer)
+        for stop_tok in prompt_template.get_stop_tokens_for_generation():
+            tok_ids = self.tokenizer.encode(stop_tok, add_special_tokens=False)
+            stop_token_ids.append(tok_ids[-1])
+
+        try:
+            sampling_params = SamplingParams(
+                n=request.n,
+                presence_penalty=request.presence_penalty,
+                frequency_penalty=request.frequency_penalty,
+                temperature=request.temperature,
+                top_p=request.top_p,
+                stop=request.stop,
+                stop_token_ids=stop_token_ids,
+                max_tokens=request.max_tokens,
+                best_of=request.best_of,
+                top_k=request.top_k,
+                ignore_eos=request.ignore_eos,
+                use_beam_search=request.use_beam_search,
+                skip_special_tokens=False,
+                logprobs=len(self.tokenizer.vocab.keys()),
+            )
+        except ValueError as e:
+            return create_error_response(HTTPStatus.BAD_REQUEST, str(e))
+        
+
+        if GRAMMAR_SAMPLING:
+            result_generator = self.model.generate(
+                prompt=None,
+                sampling_params=sampling_params,
+                request_id=request_id,
+                prompt_token_ids=prompt_token_ids,
+                tools_or_functions=tools_or_functions,
+                prompt_template_cls=prompt_template,
+                tool_choice=request.tool_choice,
+            )
+        else:
+            result_generator = self.model.generate(
+                prompt=None,
+                sampling_params=sampling_params,
+                request_id=request_id,
+                prompt_token_ids=prompt_token_ids,
+            )
+        
```

Comment:
Seems like streaming is not supported yet. Maybe we should `assert request.stream == False` here?

Reply:
added checks for streaming requests and also added the assert